src/web/backend/authentication/authentication.controller.ts

```diff
@@ -28,18 +32,68 @@ class AuthenticationController implements Controller {
         this.router.post(`${this.path}/register`, this.registration);
         this.router.post(`${this.path}/login`, this.login);
         this.router.post(`${this.path}/isAuthenticated`, this.isAuthenticated);
+        this.router.post(`${this.path}/getUser`, this.getUser);
+        this.router.get(`${this.path}/verify`, this.verify);
+        this.router.get(
+            `${this.path}/google`,
+            passport.authenticate("google", { session: false, scope: ["profile", "email"]}));
+        this.router.get(
+            `${this.path}/google/callback`,
+            passport.authenticate('google', { session: false, failureRedirect: '/' }),
+            this.googleCallback);
     }
 
-    private createToken(user: DixitUser) {
+    private googleCallback = async (request: express.Request, response: express.Response, next: express.NextFunction) => {
+        const profile = request.user as Profile;
+        let user = await getRepository(DixitUser).findOne({email: profile._json.email});
+
+        if(!user) {
+            // TODO create form to enter nickname
+            // forbid creating nicknames that already exists
+            try {
+                user = await this.authenticationService.registerGoogle(profile);
+            } catch (e) {
+                console.log(e)
+                response.redirect('/');
+                return;
+            }
+        }
+
+        const token = await this.createToken(user);
+
+        response.cookie('oauth_jwt_token', token, {maxAge: 1000*60*60});//1 hour cookie
```

Comment:
later you take this cookie on frontend and save it in local storage. Is it really necessary to return a cookie for Google Auth? if not - consider using a response header with token here, and just save it in storage on frontend.